File: INTERVIEW/Devsu/2_countFrequencies.js
/* 
    count the word frequency in the array and return an array of numbers, the numbers must be ordered alphabetically according to the words
*/


const amountIndex = 1

const wordIndex = 0
const sortByAlphabet = (first, second) => {
    if(first[wordIndex] > second[wordIndex]){
        return 1 
    }else{
        return -1
    }
}

const countFrequency = (arrWords) => {
    const objCounter = {}

    for(let x = 0; x < arrWords.length; x++){
        if(objCounter[arrWords[x]] === undefined){
            objCounter[arrWords[x]] = 1
        }else{
            objCounter[arrWords[x]] += 1
        }
    }
    const wordsCounted = []
    for(let key in objCounter){
        wordsCounted.push([key, objCounter[key]])
    }
    return wordsCounted.sort(sortByAlphabet).map((item) => item[amountIndex])
}

const words = ['the', 'the', 'dog', 'got', 'the', 'bone', 'bone']

const res = countFrequency(words)
res

File: INTERVIEW/Devsu/2_countFrequencies.py
def countFrecuency(arrWords):
    amount_index = 0
    word_index = 1
    words = {}

    for x in arrWords:
        if x not in words:
            words[x] = 1
        else:
            words[x] = words[x] + 1 

    words_counted = [[key, value] for key, value in words.items()]
    words_counted.sort(key=lambda item:item[word_index])

    return [item[amount_index] for item in words_counted]

words = ['the', 'the', 'dog', 'got', 'the', 'bone', 'bone']
res = countFrecuency(words)
print(res)